rust/libnewsboat-ffi/src/configpaths.rs

```diff
@@ -0,0 +1,140 @@
+use abort_on_panic;
+use libc::{c_char, c_void};
+use libnewsboat::cliargsparser::CliArgsParser;
+use libnewsboat::configpaths::ConfigPaths;
+use std::ffi::{CStr, CString};
+use std::mem;
+use std::panic::{RefUnwindSafe, UnwindSafe};
+use std::ptr;
+
+#[no_mangle]
+pub extern "C" fn create_rs_configpaths() -> *mut c_void {
+    abort_on_panic(|| Box::into_raw(Box::new(ConfigPaths::new())) as *mut c_void)
+}
+
+#[no_mangle]
+pub extern "C" fn destroy_rs_configpaths(object: *mut c_void) {
+    abort_on_panic(|| {
+        if object.is_null() {
+            return;
+        };
+        unsafe {
+            Box::from_raw(object);
```

Comment:
Another comment from irc by @jebrosen was that this needs to be `Box::from_raw(object as *mut ConfigPaths)` with the concrete type specified. Otherwise rustc is going to infer this as `Box<c_void>` and deallocate incorrectly.